engine/engine.go

```diff
@@ -250,43 +257,69 @@ func (e *Engine) PublishLatest(ctx context.Context) (cid.Cid, error) {
 	return adCid, nil
 }
 
-// PublishLatestHTTP publishes the latest existing advertisement to a specific
-// indexer.
-func (e *Engine) PublishLatestHTTP(ctx context.Context, indexerHost string) (cid.Cid, error) {
+// PublishLatestHTTP publishes the latest existing advertisement to the
+// specific indexers.
+func (e *Engine) PublishLatestHTTP(ctx context.Context, announceURLs ...*url.URL) (cid.Cid, error) {
 	adCid, err := e.latestAdToPublish(ctx)
 	if err != nil {
 		return cid.Undef, err
 	}
 
+	err = e.httpAnnounce(ctx, adCid, announceURLs)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return adCid, nil
+}
+
+func (e *Engine) httpAnnounce(ctx context.Context, adCid cid.Cid, announceURLs []*url.URL) error {
 	ai := &peer.AddrInfo{
 		ID: e.h.ID(),
 	}
 
 	switch e.pubKind {
 	case NoPublisher:
 		log.Info("Remote announcements disabled")
-		return cid.Undef, nil
+		return nil
 	case DataTransferPublisher:
 		ai.Addrs = e.h.Addrs()
 	case HttpPublisher:
 		maddr, err := hostToMultiaddr(e.pubHttpListenAddr)
 		if err != nil {
-			return cid.Undef, err
+			return err
 		}
 		proto, _ := multiaddr.NewMultiaddr("/http")
 		ai.Addrs = append(ai.Addrs, multiaddr.Join(maddr, proto))
 	}
 
-	cl, err := httpclient.New(indexerHost)
-	if err != nil {
-		return cid.Undef, err
+	errChan := make(chan error)
+	for _, u := range announceURLs {
+		// Send HTTP announce to indexers concurrently.
+		go func(announceURL *url.URL) {
+			log.Infow("Announcing advertisement over HTTP", "url", announceURL)
+			cl, err := httpclient.New(announceURL.String())
+			if err != nil {
+				errChan <- fmt.Errorf("failed to create http client for indexer %s: %w", announceURL, err)
+				return
+			}
+			err = cl.Announce(ctx, ai, adCid)
+			if err != nil {
+				errChan <- fmt.Errorf("failed to send http announce to indexer %s: %w", announceURL, err)
+				return
+			}
+			errChan <- nil
+		}(u)
 	}
 
-	if err = cl.Announce(ctx, ai, adCid); err != nil {
-		return cid.Undef, err
+	var errs error
+	for i := 0; i < len(announceURLs); i++ {
```

Comment:
`select` on `errChan` and `ctx`? We have context in this function already. Would be good to use it in case context has a timeout.

Reply:
A select is not needed because `ctx` is passed into `cl.Announce`.  If `ctx` is canceled then all the pending requests will be canceled and return the canceled error.  That will cause the `errChan` reader to collect those errors and return.

Reply:
That's fine. 

One could argue that if `ctx` is cancelled, then no announce should be called to begin with since there is no point calling an announce. 

Marking as non blocker.